src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1482 @@
+        content_creator_name: Union[str, pydicom.valuerep.PersonName, None], optional
+            Name of the person who created the content of this presentation
+            state.
+        content_creator_identification: Union[highdicom.ContentCreatorIdentificationCodeSequence, None], optional
+            Identifying information for the person who created the content of
+            this presentation state.
+        rescale_intercept: Union[int, float, None], optional
+            Intercept used for rescaling pixel values.
+        rescale_slope: Union[int, float, None], optional
+            Slope used for rescaling pixel values.
+        rescale_type: Union[highdicom.RescaleTypeValues, str, None], optional
+            String or enumerated value specifying the units of the output of
+            the Modality LUT or rescale operation.
+        modality_lut: Union[highdicom.ModalityLUT, None], optional
+            Lookup table specifying a pixel rescaling operation to apply to
+            the stored values to give modality values.
+        copy_modality_lut: bool, optional
+            Include elements of the Modality LUT module (including
+            RescaleIntercept, RescaleSlope and ModalityLUTSequence), if any,
+            in the presentation state with values copied from the source images.
+        softcopy_voi_luts: Union[Sequence[highdicom.pr.SoftcopyVOILUT], None], optional
+            One or more pixel value-of-interest operations to be applied after
+            the modality LUT and/or rescale operation.
+        copy_voi_lut: bool, optional
+            Include elements of the Softcopy VOI LUT module (including
+            WindowWidth, WindowCenter, and VOILUTSequence), if any, in the
+            presentation state with values copied from the source images.
+        icc_profile: Union[bytes, None], optional
+            ICC color profile object to include in the presentation state. If
+            none is provided, a standard RGB ("sRGB") profile will be assumed.
+            The profile must follow the constraints listed in
+            :dcm:`C.11.15 <part03/sect_C.11.15.html>`.
+        transfer_syntax_uid: Union[str, highdicom.UID], optional
+            Transfer syntax UID of the presentation state.
+        **kwargs: Any, optional
+            Additional keyword arguments that will be passed to the constructor
+            of `highdicom.base.SOPClass`
+
+        """  # noqa: E501
+        # Check referenced images are from the same series and have the same
+        # size
+        ref_series_uid = referenced_images[0].SeriesInstanceUID
+        ref_im_rows = referenced_images[0].Rows
+        ref_im_columns = referenced_images[0].Columns
+        for ref_im in referenced_images:
+            series_uid = ref_im.SeriesInstanceUID
+            if series_uid != ref_series_uid:
+                raise ValueError(
+                    'Images belonging to different series are not supported.'
```

Comment:
```suggestion
                    'All referenced images must belong to the same series.'
```

Reply:
I have accepted the suggestion but the phrasing was like this because I don't think there is any reason in principle why images from different series could be used, I just chose not to support that case (yet).